frontend/src/component/feature/FeatureStrategy/FeatureStrategyConstraints/EditableConstraint/useEditableConstraint/editable-constraint-type.test.ts

```diff
@@ -0,0 +1,44 @@
+import { createEmptyConstraint } from 'utils/createEmptyConstraint';
+import { fromIConstraint, toIConstraint } from './editable-constraint-type.ts';
+import { constraintId } from 'constants/constraintId';
+
+test('mapping to and from retains the constraint id', () => {
+    const constraint = createEmptyConstraint('context');
+
+    expect(toIConstraint(fromIConstraint(constraint))[constraintId]).toEqual(
+        constraint[constraintId],
+    );
+});
+
+test('mapping to an editable constraint adds a constraint id if there is none', () => {
+    const constraint = createEmptyConstraint('context');
+    delete constraint[constraintId];
+
+    const editableConstraint = fromIConstraint(constraint);
+
+    expect(editableConstraint[constraintId]).toBeDefined();
+
+    const iConstraint = toIConstraint(editableConstraint);
+    expect(iConstraint[constraintId]).toEqual(editableConstraint[constraintId]);
+});
+
+test('mapping from an empty constraint removes redundant value / values', () => {
+    const constraint = createEmptyConstraint('context');
+    expect('value' in constraint).toBe(true);
```

Comment:
is there toHaveProperty assertion?

Reply:
huh. didn't think of that, but yes, there is. Good shout.